src/parser/type_checking.c

```diff
@@ -13232,7 +13232,7 @@ pt_eval_function_type (PARSER_CONTEXT * parser, PT_NODE * node)
 	    found_supported = 0;
 	    for (i = 0; i < len; i++)
 	      {
-		if (arg->type_enum == current_types[i])
+		if (arg->type_enum == current_types[i] || PT_IS_STRING_TYPE (arg->type_enum))
```

Comment:
please add inline function in parse_tree.h in pt_is_json_value_type; function should include switch with accepted types and default for not accepted types.

you can then use it instead of:
```c
	    for (i = 0; i < len; i++)
	      {
		if (supported_types[i] == arg->type_enum)
		  {
		    found_supported_type = 1;
		    break;
		  }
	      }
```

Same goes for the other type_checking cases too.

Reply:
I support Remzi :)